File: src/ZKWeb.MVVMPlugins/MVVM.Common.Organization/src/Application/Services/RoleManageService.cs
using AutoMapper;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using ZKWeb.MVVMPlugins.MVVM.Common.Base.src.Application.Dtos;
using ZKWeb.MVVMPlugins.MVVM.Common.Base.src.Application.Services.Bases;
using ZKWeb.MVVMPlugins.MVVM.Common.Organization.src.Application.Dtos;
using ZKWeb.MVVMPlugins.MVVM.Common.Organization.src.Components.ActionFilters;
using ZKWeb.MVVMPlugins.MVVM.Common.Organization.src.Domain.Entities;
using ZKWeb.MVVMPlugins.MVVM.Common.Organization.src.Domain.Entities.Interfaces;
using ZKWeb.MVVMPlugins.MVVM.Common.Organization.src.Domain.Services;
using ZKWebStandard.Ioc;

namespace ZKWeb.MVVMPlugins.MVVM.Common.Organization.src.Application.Services {
	/// <summary>
	/// 角色管理服务
	/// </summary>
	[ExportMany, SingletonReuse, Description("角色管理服务")]
	public class RoleManageService : ApplicationServiceBase {
		private RoleManager _roleManager;

		public RoleManageService(RoleManager roleManager) {
			_roleManager = roleManager;
		}

		[Description("搜索角色")]
		[CheckPrivilege(true, typeof(IAmAdmin), "Role:View")]
		public GridSearchResponseDto Search(GridSearchRequestDto request) {
			return request.BuildResponse<Role, Guid>()
				.FilterKeywordWith(t => t.Name)
				.FilterKeywordWith(t => t.Remark)
				.ToResponse<RoleOutputDto>();
		}

		[Description("编辑角色")]
		[CheckPrivilege(true, typeof(IAmAdmin), "Role:Edit")]
		public ActionResponseDto Edit(RoleInputDto dto) {
			var role = _roleManager.Get(dto.Id) ?? new Role();
			Mapper.Map(dto, role);
			_roleManager.Save(ref role);
			return ActionResponseDto.CreateSuccess("Saved Successfully");
		}

		[Description("删除角色")]
		[CheckPrivilege(true, typeof(IAmAdmin), "Role:Remove")]
		public ActionResponseDto Remove(Guid id) {
			_roleManager.Delete(id);
			return ActionResponseDto.CreateSuccess("Deleted Successfully");
		}

		[Description("获取所有角色")]
		public IList<RoleOutputDto> GetAllRoles() {
			var roles = _roleManager.GetMany();
			return roles.Select(r => Mapper.Map<RoleOutputDto>(r)).ToList();
		}
	}
}

File: src/ZKWeb.MVVMPlugins/MVVM.Angular.Website/static.old/src/modules/generated_module/services/role-manage-service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { AppApiService } from '../../base_module/services/app-api-service';
import { GridSearchResponseDto } from '../dtos/grid-search-response-dto';
import { GridSearchRequestDto } from '../dtos/grid-search-request-dto';
import { ActionResponseDto } from '../dtos/action-response-dto';
import { RoleInputDto } from '../dtos/role-input-dto';
import { RoleOutputDto } from '../dtos/role-output-dto';

@Injectable()
/** 角色管理服务 */
export class RoleManageService {
    constructor(private appApiService: AppApiService) { }

    /** 搜索角色 */
    Search(request: GridSearchRequestDto): Observable<GridSearchResponseDto> {
        return this.appApiService.call<GridSearchResponseDto>(
            "/api/RoleManageService/Search",
            {
                request
            });
    }

    /** 编辑角色 */
    Edit(dto: RoleInputDto): Observable<ActionResponseDto> {
        return this.appApiService.call<ActionResponseDto>(
            "/api/RoleManageService/Edit",
            {
                dto
            });
    }

    /** 删除角色 */
    Remove(id: string): Observable<ActionResponseDto> {
        return this.appApiService.call<ActionResponseDto>(
            "/api/RoleManageService/Remove",
            {
                id
            });
    }

    /** 获取所有角色 */
    GetAllRoles(): Observable<RoleOutputDto[]> {
        return this.appApiService.call<RoleOutputDto[]>(
            "/api/RoleManageService/GetAllRoles",
            {
            });
    }
}
